src/components/categories/Edit.vue

```diff
@@ -0,0 +1,214 @@
+<script setup lang="ts">
+import { useAsyncState } from '@vueuse/core'
+import { computed, ref } from 'vue'
+import { onBeforeRouteLeave, useRouter } from 'vue-router'
+import { useApi } from '../../composables/useApi'
+import { useAuth } from '../../composables/useAuth'
+import { useSessionToken } from '../../composables/useSessionToken'
+import { Categories } from '../../lib/api/Categories'
+import {
+	ProblemDetails,
+	UpdateCategoryRequest
+} from '../../lib/api/data-contracts'
+import { HttpResponse } from '../../lib/api/http-client'
+import Slug from '../blocks/Slug.vue'
+
+const props = defineProps<{
+	id: number
+}>()
+
+const updateError = ref('')
+
+const token = useSessionToken()
+
+const router = useRouter()
+
+const categories = new Categories({
+	baseUrl: import.meta.env.VITE_BACKEND_URL
+})
+
+const warnBeforeLeave = ref(true)
+
+const updateRequest = ref<UpdateCategoryRequest>({})
+
+const {
+	state: cat,
+	error,
+	isLoading,
+	execute
+} = useAsyncState(async () => {
+	const resp = await categories.getCategory(props.id)
+	updateRequest.value.info = resp.data.info || ''
+	updateRequest.value.name = resp.data.name
+	updateRequest.value.slug = resp.data.slug
+	updateRequest.value.sortDirection = resp.data.sortDirection
+	return resp.data
+}, null)
+
+onBeforeRouteLeave(() => {
+	if (
+		warnBeforeLeave.value &&
+		(cat.value?.name !== updateRequest.value.name ||
+			cat.value?.slug !== updateRequest.value.slug ||
+			cat.value?.info !== updateRequest.value.info ||
+			cat.value?.sortDirection !== updateRequest.value.sortDirection)
+	) {
+		if (!window.confirm('Do you want to leave? You have unsaved changes.')) {
+			return false
+		}
+	}
+})
+
+const errorResponse = computed(
+	() => error.value as HttpResponse<unknown, void | ProblemDetails>
+)
+
+async function submit() {
+	useApi(
+		() =>
+			categories.updateCategory(
+				props.id,
+				{
+					name: updateRequest.value.name,
+					info: updateRequest.value.info,
+					slug: updateRequest.value.slug,
+					sortDirection: updateRequest.value.sortDirection
+				},
+				useAuth(token.value)
+			),
+		() => {
+			warnBeforeLeave.value = false
+			router.push({ name: 'categoryView', params: { id: props.id } })
+		},
+		(error) => {
+			updateError.value = 'Failed to update: ' + error.status.toString(10)
+		}
+	)
+}
+</script>
+
+<template>
+	<div class="container">
+		<div v-if="isLoading">Loading...</div>
+		<div v-else-if="error" class="error-container">
+			<p class="errorText">
+				<!-- For unexpected server errors; fields will be empty, hence the defaults. -->
+				Failed to fetch category: {{ errorResponse.status }}
+				{{ errorResponse.error?.title ?? 'Check the console for more info.' }}
```

Comment:
I think something like "unexpected error" or "unknown error" would be more appropriate here. I would imagine some users don't even know how to open the developer console to begin with, and using this application shouldn't require interacting with it.